editor/src/core/shared/array-utils.ts

```diff
@@ -327,7 +327,7 @@ export function immutablyUpdateArrayIndex<T>(
   return working
 }
 
-export function safeIndex<T>(array: Array<T>, index: number): T | undefined {
+export function safeIndex<T>(array: Array<T> | ReadonlyArray<T>, index: number): T | undefined {
```

Comment:
I believe this can just be ReadonlyArray here
```suggestion
export function safeIndex<T>(array: ReadonlyArray<T>, index: number): T | undefined {
```

Reply:
Yep, fixed 👍